components/inbound-endpoints/org.wso2.carbon.inbound.endpoint/src/main/java/org/wso2/carbon/inbound/endpoint/protocol/rabbitmq/RabbitMQInjectHandler.java

```diff
@@ -139,6 +139,14 @@ public boolean invoke(RabbitMQMessage message) {
             log.error("Sequence: " + injectingSeq + " not found");
         }
 
+        Object o = msgCtx.getProperty(RabbitMQConstants.SET_ROLLBACK_ONLY);
+        if (o != null) {
```

Comment:
There are two situations invoke() method return false.
1. injectingSeq is null or empty
2. SET_ROLLBACK_ONLY property set to true
Therefore, I think developer may get confused if we use last return statement based SET_ROLLBACK_ONLY because it's only used in transactions. How about below change? Thoughts?

```
    public boolean invoke(RabbitMQMessage message) {

        boolean success = true;
        // rest of the code

        if (injectingSeq == null || injectingSeq.equals("")) {
            log.error("Sequence name not specified. Sequence : " + injectingSeq);
            success = false;
        }
        //rest of the code

        Object property = msgCtx.getProperty(RabbitMQConstants.SET_ROLLBACK_ONLY);
        if (property != null && (property instanceof Boolean && ((Boolean) property))
                    || (property instanceof String && Boolean.valueOf((String) property))) {
                success = false;
        }

        return success;
    }
```